android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -521,6 +521,28 @@ class GeckoEngineSession(
         geckoSession.setPriorityHint(priority.id)
     }
 
+    /**
+     * See [EngineSession.checkForFormData].
+     */
+    override fun checkForFormData() {
+        geckoSession.containsFormData().then(
+            { result ->
+                if (result == null) {
+                    logger.error("No result from GeckoView containsFormData.")
```

Comment:
is this an error that we would want to measure the telemetries as well? What does `null` indicate here about the containsFormData bool?

Reply:
It's true we could probably measure this event as well. To be honest I don't think the telemetry will be all that useful, it seems unlikely to me that GeckoView will return null or even throw. We could remove this, but this form data feature is just to test and see if it will have a significant impact on users who notice their tabs being killed more often. Perhaps we could add more telemetry in the future.

Reply:
yes agreed, this should be a next step in case it is worth it :)